snakemake/notebook.py

```diff
@@ -42,12 +42,12 @@ def draft(self, listen):
 
         nb["cells"].append(nbformat.v4.new_code_cell("# start coding here"))
 
-        os.makedirs(os.path.dirname(self.local_path), exist_ok=True)
+        os.makedirs(os.path.dirname(str(self.local_path)), exist_ok=True)
```

Comment:
Can you use `self.local_path.get_path_or_uri()` here? That is more idiomatic with the new API.

Reply:
And in all str() occurrences below as well.